common/network.c

```diff
@@ -417,3 +417,22 @@ network_routing_rules_set_all_main(bool flush)
 				      "lookup", IP_ROUTING_TABLE, NULL };
 	return proc_fork_and_execvp(argv2);
 }
+
+list_t *
+network_get_physical_interfaces_new()
+{
+	struct if_nameindex *if_ni, *i;
+	if_ni = if_nameindex();
+
+	list_t *if_name_list = NULL;
+
+	IF_NULL_RETVAL(if_ni, NULL);
+
+	for (i = if_ni; i->if_index != 0 || i->if_name != NULL; i++) {
+		char *dev_drv_path = mem_printf("/sys/class/net/%s/device/driver", i->if_name);
+		if (file_exists(dev_drv_path))
+			if_name_list = list_append(if_name_list, mem_strdup(i->if_name));
```

Comment:
if_index == 0 && if_name != NULL possible?
(would strdup on NULL)

Reply:
there is no netif without a name